### PR TITLE
Update README.md to improve setting up dev env

### PR DESCRIPTION
**Pull request by cibic89**
Contributions:
- update README

### CODE REVIEW

**Comment by cibic89** [comment]
@marctorsoc

File: README.md [hunk 1]
```diff
@@ -51,42 +51,97 @@ print(space_eval(space, best))
...
-$ git clone https://github.com/hyperopt/hyperopt.git
+$ git clone https://github.com/<github username>/hyperopt.git
 $ cd hyperopt
+```
+
+### 4. Create a virtual env (python 3.x) to contain the dependencies and activate it
+on Windows using virtualenv:  
+`$ python -m venv venv`  
+or with conda  
+`$ conda create --name hyperopt`
```
Comment (marctorsoc) [inline README.md:73]: `conda create -n my_env python=3`

Comment (marctorsoc) [reply README.md:73]: and it'd be great to change `venv` on the rest to `my_env` to make it more clear

File: README.md [hunk 2]
```diff
@@ -51,42 +51,97 @@ print(space_eval(space, best))
...
+`$ python -m venv venv`  
+or with conda  
+`$ conda create --name hyperopt`  
+or Linux/Unix  
+`$ python3 -m venv venv`
 
-# Create a virtual env (python 3.x) to contain the dependencies and activate it
-$ python3 -m venv venv
+```bash
 $ source venv/bin/activate
```
Comment (marctorsoc) [inline README.md:78]: or `conda activate my_env`

Comment (marctorsoc) [reply README.md:78]: and add a text before please to explain what we're doing

File: README.md [hunk 3]
```diff
@@ -51,42 +51,97 @@ print(space_eval(space, best))
...
+$ python setup.py &&  pip install -e
```
Comment (marctorsoc) [inline README.md:79]: `python setup.py` alone will do nothing, it needs a command. Could you try just with the `pip install -e .....` ? I think it's not needed and the `pip install ....` will be enough

File: README.md [hunk 4]
```diff
@@ -51,42 +51,97 @@ print(space_eval(space, best))
...
-We recommend to use [Black](https://github.com/psf/black) to format your code before submitting a PR. You can use it 
-with a pre-commit hook as follows:
+and start making changes. Always use a feature branch. It’s good practice to never work on the master branch!
+
+8. Develop the feature on your feature branch on your computer, using
+   Git to do the version control. When you’re done editing, add changed
+   files using git add and then git commit:
+```bash
+$ git add modified_files
+$ git commit "my first hyperopt commit"
```
Comment (marctorsoc) [inline README.md:111]: `git commit -m ...`

File: README.md [hunk 5]
```diff
@@ -51,42 +51,97 @@ print(space_eval(space, best))
...
+3. Clone your fork of the hyperopt repo from your GitHub account to your
+   local disk:
 
 ```bash
-$ git clone https://github.com/hyperopt/hyperopt.git
+$ git clone https://github.com/<github username>/hyperopt.git
 $ cd hyperopt
+```
+
+### 4. Create a virtual env (python 3.x) to contain the dependencies and activate it
```
Comment (marctorsoc) [inline README.md:69]: I think better that the sections group different steps (like Formatting). Here we have a section for one step, and the github part is part of this right now, which doesn't make a lot of sense. Could you organise a bit better the doc?

File: README.md (hunk 3 above)
Comment (cibic89) [reply README.md:79]: It's not, it needs this `python setup.py develop && pip install -e '.[MongoTrials, SparkTrials, ATPE, dev]'`

**Comment by marctorsoc** [comment]
the next PR that you were going to do is already here https://github.com/hyperopt/hyperopt/pull/697. Given that, we can reframe this PR to simply updating the README

**Comment by cibic89** [comment]
did the changes 👍

File: README.md [hunk 6]
```diff
@@ -67,19 +70,26 @@ $ cd hyperopt
...
+then install:
 ```bash
-$ source venv/bin/activate
-$ python setup.py &&  pip install -e
-'.[MongoTrials, SparkTrials, ATPE, dev]'`
+$ python setup.py develop
+$ pip install -e .[MongoTrials]
+$ pip install -e .[SparkTrials]
+$ pip install -e .[ATPE]
+$ pip install -e .[dev]
```
Comment (marctorsoc) [inline README.md:89]: this is not needed, please change to `pip install -e '.....'` only

File: README.md (hunk 5 above)
Comment (marctorsoc) [reply README.md:72]: bump

File: README.md (hunk 3 above)
Comment (marctorsoc) [reply README.md]: I tested locally, pip install is enough

**Comment by marctorsoc** [comment]
> LGTM. Added just some more comments to the README. Please make sure that the tests pass. Unfortunately some our tests are not stable (very welcome to fix), so you might need to commit again to get a proper seed and pass the tests. You can run the tests locally first to make sure everything works before using the CI

make sure tests pass @cibic89

File: README.md (hunk 6 above)
Comment (cibic89) [reply README.md:89]: have you tested your assumption? I found that you cannot install multiple extras in one command using pip. The only way to guarantee the extras install is to use a single line. Tried the following:

```bash
pip install -e '.[MongoTrials, SparkTrials, ATPE, dev]'
pip install -e .[MongoTrials, SparkTrials, ATPE, dev]
pip install -e .["MongoTrials", "SparkTrials", "ATPE", "dev"]
pip install -e '.["MongoTrials", "SparkTrials", "ATPE", "dev"]'
```

of which none work consistently.

File: README.md (hunk 5 above)
Comment (cibic89) [reply README.md:72]: ditto, pushing in a minute

File: README.md (hunk 6 above)
Comment (marctorsoc) [reply README.md:89]: this worked for me `pip install -e '.[ATPE, dev]'`. I just tested. I can imagine that with 4 should work as well. Can you test:
1. create a fresh environment
2. do `pip install -e '.[ATPE, dev]'`
3. check that black and lightgbm are installed (they come each with one of those extras)

If this doesn't work might be because of Windows... but seems weird to me tbh

File: README.md (hunk 3 above)
Comment (cibic89) [reply README.md]: > 
> 
> I tested locally, pip install is enough

Got it, trusting you on this as this doesn't work in Windows.

File: README.md (hunk 6 above)
Comment (cibic89) [reply README.md:89]: > 
> 
> > LGTM. Added just some more comments to the README. Please make sure that the tests pass. Unfortunately some our tests are not stable (very welcome to fix), so you might need to commit again to get a proper seed and pass the tests. You can run the tests locally first to make sure everything works before using the CI
> 
> make sure tests pass @cibic89

This test fails on both Windows and Ubuntu with the same error:
```
hyperopt/tests/test_atpe_basic.py Fatal Python error: Aborted

Thread 0x00007fad58bab700 (most recent call first):
  File "/home/george/miniconda3/lib/python3.7/threading.py", line 300 in wait
  File "/home/george/miniconda3/lib/python3.7/threading.py", line 552 in wait
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/tqdm-4.48.0-py3.7.egg/tqdm/_monitor.py", line 69 in run
  File "/home/george/miniconda3/lib/python3.7/threading.py", line 926 in _bootstrap_inner
  File "/home/george/miniconda3/lib/python3.7/threading.py", line 890 in _bootstrap

Current thread 0x00007fad62221740 (most recent call first):
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/lightgbm/basic.py", line 1744 in __init__
  File "/home/george/code_projects/hyperopt/hyperopt/atpe.py", line 682 in __init__
  File "/home/george/code_projects/hyperopt/hyperopt/atpe.py", line 1575 in suggest
  File "/home/george/code_projects/hyperopt/hyperopt/fmin.py", line 269 in run
  File "/home/george/code_projects/hyperopt/hyperopt/fmin.py", line 352 in exhaust
  File "/home/george/code_projects/hyperopt/hyperopt/fmin.py", line 539 in fmin
  File "/home/george/code_projects/hyperopt/hyperopt/tests/test_atpe_basic.py", line 28 in test_run_basic_search
  File "/home/george/miniconda3/lib/python3.7/unittest/case.py", line 628 in run
  File "/home/george/miniconda3/lib/python3.7/unittest/case.py", line 676 in __call__
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/unittest.py", line 231 in runtest
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 135 in pytest_runtest_call
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 217 in <lambda>
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 244 in from_call
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 217 in call_runtest_hook
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 186 in call_and_report
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 100 in runtestprotocol
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 85 in pytest_runtest_protocol
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/main.py", line 272 in pytest_runtestloop
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/main.py", line 247 in _main
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/main.py", line 191 in wrap_session
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/main.py", line 240 in pytest_cmdline_main
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 125 in main
  File "/home/george/code_projects/hyperopt/venv/bin/pytest", line 8 in <module>
Aborted (core dumped)

```

Comment (cibic89) [reply README.md:89]: > 
> 
> this worked for me `pip install -e '.[ATPE, dev]'`. I just tested. I can imagine that with 4 should work as well. Can you test:
> 
>     1. create a fresh environment
> 
>     2. do `pip install -e '.[ATPE, dev]'`
> 
>     3. check that black and lightgbm are installed (they come each with one of those extras)
> 
> 
> If this doesn't work might be because of Windows... but seems weird to me tbh

You are spot on; it was failing because of Windows but don't have this issue any more. Switched to Ubuntu.

**Comment by cibic89** [comment]
> 
> 
> > LGTM. Added just some more comments to the README. Please make sure that the tests pass. Unfortunately some our tests are not stable (very welcome to fix), so you might need to commit again to get a proper seed and pass the tests. You can run the tests locally first to make sure everything works before using the CI
> 
> make sure tests pass @cibic89

hyperopt/tests/test_atpe_basic.py Fatal Python error: Aborted

Thread 0x00007fad58bab700 (most recent call first):
  File "/home/george/miniconda3/lib/python3.7/threading.py", line 300 in wait
  File "/home/george/miniconda3/lib/python3.7/threading.py", line 552 in wait
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/tqdm-4.48.0-py3.7.egg/tqdm/_monitor.py", line 69 in run
  File "/home/george/miniconda3/lib/python3.7/threading.py", line 926 in _bootstrap_inner
  File "/home/george/miniconda3/lib/python3.7/threading.py", line 890 in _bootstrap

Current thread 0x00007fad62221740 (most recent call first):
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/lightgbm/basic.py", line 1744 in __init__
  File "/home/george/code_projects/hyperopt/hyperopt/atpe.py", line 682 in __init__
  File "/home/george/code_projects/hyperopt/hyperopt/atpe.py", line 1575 in suggest
  File "/home/george/code_projects/hyperopt/hyperopt/fmin.py", line 269 in run
  File "/home/george/code_projects/hyperopt/hyperopt/fmin.py", line 352 in exhaust
  File "/home/george/code_projects/hyperopt/hyperopt/fmin.py", line 539 in fmin
  File "/home/george/code_projects/hyperopt/hyperopt/tests/test_atpe_basic.py", line 28 in test_run_basic_search
  File "/home/george/miniconda3/lib/python3.7/unittest/case.py", line 628 in run
  File "/home/george/miniconda3/lib/python3.7/unittest/case.py", line 676 in __call__
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/unittest.py", line 231 in runtest
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 135 in pytest_runtest_call
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 217 in <lambda>
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 244 in from_call
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 217 in call_runtest_hook
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 186 in call_and_report
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 100 in runtestprotocol
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 85 in pytest_runtest_protocol
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/main.py", line 272 in pytest_runtestloop
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/main.py", line 247 in _main
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/main.py", line 191 in wrap_session
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/main.py", line 240 in pytest_cmdline_main
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 125 in main
  File "/home/george/code_projects/hyperopt/venv/bin/pytest", line 8 in <module>
Aborted (core dumped)

File: README.md (hunk 6 above)
Comment (marctorsoc) [reply README.md:89]: we should give instructions that work on Windows as well. So maybe do two (sets of) commands: one for Windows users, and one for the rest, as for the virtualenv. Eventually we can revisit this to try to generate instructions that are short and work for both...

**Comment by marctorsoc** [comment]
> > > LGTM. Added just some more comments to the README. Please make sure that the tests pass. Unfortunately some our tests are not stable (very welcome to fix), so you might need to commit again to get a proper seed and pass the tests. You can run the tests locally first to make sure everything works before using the CI
> > 
> > 
> > make sure tests pass @cibic89
> 
> ```
> hyperopt/tests/test_atpe_basic.py Fatal Python error: Aborted
> 
> Thread 0x00007fad58bab700 (most recent call first):
>   File "/home/george/miniconda3/lib/python3.7/threading.py", line 300 in wait
>   File "/home/george/miniconda3/lib/python3.7/threading.py", line 552 in wait
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/tqdm-4.48.0-py3.7.egg/tqdm/_monitor.py", line 69 in run
>   File "/home/george/miniconda3/lib/python3.7/threading.py", line 926 in _bootstrap_inner
>   File "/home/george/miniconda3/lib/python3.7/threading.py", line 890 in _bootstrap
> 
> Current thread 0x00007fad62221740 (most recent call first):
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/lightgbm/basic.py", line 1744 in __init__
>   File "/home/george/code_projects/hyperopt/hyperopt/atpe.py", line 682 in __init__
>   File "/home/george/code_projects/hyperopt/hyperopt/atpe.py", line 1575 in suggest
>   File "/home/george/code_projects/hyperopt/hyperopt/fmin.py", line 269 in run
>   File "/home/george/code_projects/hyperopt/hyperopt/fmin.py", line 352 in exhaust
>   File "/home/george/code_projects/hyperopt/hyperopt/fmin.py", line 539 in fmin
>   File "/home/george/code_projects/hyperopt/hyperopt/tests/test_atpe_basic.py", line 28 in test_run_basic_search
>   File "/home/george/miniconda3/lib/python3.7/unittest/case.py", line 628 in run
>   File "/home/george/miniconda3/lib/python3.7/unittest/case.py", line 676 in __call__
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/unittest.py", line 231 in runtest
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 135 in pytest_runtest_call
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 217 in <lambda>
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 244 in from_call
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 217 in call_runtest_hook
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 186 in call_and_report
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 100 in runtestprotocol
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 85 in pytest_runtest_protocol
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/main.py", line 272 in pytest_runtestloop
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/main.py", line 247 in _main
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/main.py", line 191 in wrap_session
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/main.py", line 240 in pytest_cmdline_main
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
>   File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 125 in main
>   File "/home/george/code_projects/hyperopt/venv/bin/pytest", line 8 in <module>
> Aborted (core dumped)
> ```

ok, nvm about the tests. You're not modifying anything tests related, so it's not your resp. to make them work. Just fix the README pip instructions, and I'll approve and merge

File: README.md (hunk 6 above)
Comment (cibic89) [reply README.md:89]: > 
> 
> we should give instructions that work on Windows as well. So maybe do two (sets of) commands: one for Windows users, and one for the rest, as for the virtualenv. Eventually we can revisit this to try to generate instructions that are short and work for both...

The single line method installs the extras on Windows but I suspect there are other problems with pytest on this platform. Let's push this for another time I think.

File: setup.py [hunk 7]
```diff
@@ -51,7 +51,7 @@
     extras_require={
         "SparkTrials": "pyspark",
         "MongoTrials": "pymongo",
-        "ATPE": ["lightgbm", "scikit-learn"],
+        "ATPE": ["lightgbm", "scikit-learn==0.22.2"],
```
Comment (marctorsoc) [inline setup.py:54]: revert this change please

Comment (cibic89) [reply setup.py]: ditto

**Comment by cibic89** [comment]
How can I help here?

**Comment by marctorsoc** [comment]
the PR is merged, there's nothing else to do :)

**Comment by cibic89** [comment]
> 
> 
> the PR is merged, there's nothing else to do :)

Meant for the sklearn 0.23+ adaptation although I see someone else started a PR

**Comment by marctorsoc** [comment]
> > the PR is merged, there's nothing else to do :)
> 
> Meant for the sklearn 0.23+ adaptation although I see someone else started a PR

yup, there's a missing test and better description. If you want to help, comment in that PR
